opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -17,4 +20,668 @@ def __init__(self, path_to_dir):
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        pol : float
+            Polarization angel of the field relative to the x plane. Can be
+            freely chosen between 0 and Pi/2 in thetaMode. For carthesian
+            coordinates it as to be either 0 or Pi/2 (x or y plane) 
+
+        m : int or str, optional
+           Only used for thetaMode geometry
+           Either 'all' (for the sum of all the modes)
+           or an integer (for the selection of a particular mode)
+
+        freq_filter : float, optional
+            Range of frequencies in percent which to filter: Frequencies higher
+            than freq_filter/100 times the dominant frequencies will be
+            filtered out
+
+        index : int or str, optional
+            Transversal index of the slice from which to calculate the envelope
+            Default is 'center', using the center slice.
+            Use 'all' to calculate a full 2D envelope
+
+        Returns
+        -------
+        A tuple with:
+        - Envelope data (1D or 2D array)
+        - A FieldMetaInformation object (note that this contains obsolete
+            information about the transversal coordinate in 1D case)
+        """
+        # Check if polarization has been entered
+        if pol is None:
+            raise ValueError('The `pol` argument is missing or erroneous.')
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Test that polarization is in x or y plane for 3D cart. coordinates
+        if self.geometry == "3dcartesian" or self.geometry == "2dcartesian":
+            if pol == 0:
+                slicing_dir = 'y'
+                coord = 'x'
+                pol = None
+            elif pol == np.pi/2:
+                slicing_dir = 'x'
+                coord = 'y'
+                pol = None
+            else:
+                raise ValueError('Only polarization in the x or y plane is '
+                                 'supported for carthesian coordinates')
+        else:
+            # Extract radial if in thetaMode
+            coord = 'r'
+            slicing_dir = None
+        # Get field data
+        field = self.get_field( t=t, iteration=iteration, field='E',
+                                coord=coord, theta=pol, m=m,
+                                slicing_dir=slicing_dir )
+        extent = field[1]
+        if index == 'center':
+            # Get central slice
+            field_slice = field[0][int( field[0].shape[0] / 2), :]
+            # Calculate inverse FFT of filtered FFT array
+            envelope = self._fft_filter(field_slice, freq_filter)
+        elif index == 'all':
+            envelope = [ self._fft_filter(field[0][i, :], freq_filter)
+                         for i in range(field[0].shape[0]) ]
+        else:
```

Comment:
I would replace 
```python
 envelope = [ self._fft_filter(field[0][i, :], freq_filter)
 for i in range(field[0].shape[0]) ]
```
by 
```
envelope = np.array( [ self._fft_filter(field[0][i, :], freq_filter)
 for i in range(field[0].shape[0]) ])
```
Because it is generally nicer to return a 2d array than a list.